src/audio/selector/selector.c

```diff
@@ -577,6 +577,8 @@ SOF_MODULE_INIT(selector, sys_comp_selector_init);
 static void build_config(struct comp_data *cd, struct module_config *cfg)
 {
 	enum sof_ipc_frame __sparse_cache frame_fmt, valid_fmt;
+	const struct sof_selector_ipc4_config sel_cfg = cd->sel_ipc4_cfg;
```

Comment:
why are you making a copy of the configuration? It doesn't seem to be modified below, so a `const` pointer would be enough